tests/tests_test.go

```diff
@@ -0,0 +1,278 @@
+				}
+			}
+		})
+
+		It("Should create a new migration", func() {
+			By("Creating a new migmigration")
+
+			migrationName := "foo"
+			migration := NewMigMigration(migrationName, migPlanName, false, false)
+			Expect(hostClient.Create(ctx, migration)).Should(Succeed())
+
+			fooMigration := &migapi.MigMigration{}
+
+			Eventually(func() string {
+				dvms := &migapi.DirectVolumeMigrationList{}
+				err := hostClient.List(ctx, dvms, client.InNamespace(MigrationNamespace))
+				if err != nil {
+					log.Println(err)
+				}
+				for _, dvm := range dvms.Items {
+					if dvm.GenerateName == migrationName+"-" {
+						if dvm.Status.Itinerary == "VolumeMigration" {
+							dvmps := &migapi.DirectVolumeMigrationProgressList{}
+							err = hostClient.List(ctx, dvmps, client.MatchingLabels{"directvolumemigration": string(dvm.UID)})
+							if err != nil {
+								log.Println(err)
+							}
+							for _, dvmp := range dvmps.Items {
+								if dvmp.Status.TotalProgressPercentage == "100%" {
+									return dvm.Status.Phase
+								}
+							}
+						} else {
+							log.Println("Direct volume migration Failed")
+						}
+					}
+				}
+				return ""
+			}, time.Minute*5, time.Second).Should(Equal("Completed"))
+			Eventually(func() string {
+				hostClient.Get(ctx, client.ObjectKey{Name: migrationName, Namespace: MigrationNamespace}, fooMigration)
+				if fooMigration.Status.Itinerary == "Final" {
+					return fooMigration.Status.Phase
+				}
+				return ""
+			}, time.Minute*5, time.Second).Should(Equal("Completed"))
+
+			// TODO check the destination pvc for data
+			Eventually(func() bool {
+				podList := &v1.PodList{}
+				hostClient.List(ctx, podList, &client.ListOptions{
+					LabelSelector: labels.SelectorFromSet(map[string]string{
+						"app": "longpvc-test",
+					},
+					),
+					Namespace: "ocp-41583-longpvcname",
+				})
+				destinationFile := []string{}
+				for _, p := range podList.Items {
+					podCmd := pods.PodCommand{
+						Pod:     &p,
+						RestCfg: hostCfg,
+						Args:    []string{"sh", "-c", "ls -l data/test"},
+					}
+					err = podCmd.Run()
+					if err != nil {
+						log.Println(err, "Failed running ls command inside destination Pod",
+							"pod", path.Join(p.Namespace, p.Name),
+							"command", "ls -l data/test")
+					}
+					destinationFile = strings.Split(podCmd.Out.String(), " ")
+				}
+				podList, err = sourceClient.CoreV1().Pods("ocp-41583-longpvcname").List(ctx, metav1.ListOptions{
+					LabelSelector: labels.SelectorFromSet(map[string]string{
+						"app": "longpvc-test",
+					},
+					).String(),
+				})
+				for _, p := range podList.Items {
+					podCmd := pods.PodCommand{
+						Pod:     &p,
+						RestCfg: sourceCfg,
+						Args:    []string{"sh", "-c", "ls -l data/test"},
+					}
+					err = podCmd.Run()
+					if err != nil {
+						log.Println(err, "Failed running ls command inside source Pod",
+							"pod", path.Join(p.Namespace, p.Name),
+							"command", "ls data/test")
```

Comment:
NIT: "data/test" is being used multiple times, could use an appropriate constant